lib/SPIRV/libSPIRV/spirv.hpp

```diff
@@ -65,6 +65,7 @@ enum SourceLanguage {
     SourceLanguageOpenCL_C = 3,
     SourceLanguageOpenCL_CPP = 4,
     SourceLanguageHLSL = 5,
+    SourceLanguageCPP_for_OpenCL = 6,
```

Comment:
There is no such value in the spec: https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_source_language_a_source_language

What is the difference between `SourceLanguageCPP_for_OpenCL` and `SourceLanguageOpenCL_CPP` ?

Reply:
I suppose the value will be added to the spec soon, as https://github.com/KhronosGroup/SPIRV-Headers/pull/217 was recently merged into the SPIR-V headers.

> What is the difference between SourceLanguageCPP_for_OpenCL and SourceLanguageOpenCL_CPP ?

C++ for OpenCL and OpenCL C++ are different languages.  Quoting from https://github.com/KhronosGroup/OpenCL-Guide/blob/main/chapters/cpp_for_opencl.md :
> The OpenCL working group has transitioned from the original OpenCL C++ kernel language first defined in OpenCL 2.2 to the community developed C++ for OpenCL kernel language that provides improved features and compatibility with OpenCL C.

Reply:
Yes, the value is being added retrospectively to all versions of SPIR-V.  When SPIR-V Tools is built with the current SPIR-V Headers, use of the CPP_for_OpenCL language ID will pass validation, and will correctly assemble and disassemble.

It was agreed that it is appropriate to add source language IDs retrospectively.  As the language ID has no semantic content, the use of unknown languages must not cause SPIR-V to fail to be consumed in an implementation.  The SPIR-V Tools will be changed to allow disassembly and assembly of modules using unknown language IDs, using a numeric form for the ID, and the validation will be relaxed to permit unknown language IDs.  For the time being, the overly strict behavior of SPIR-V Tools should not affect those who are translating from code written in e.g. OpenCL C.